oem/ibm/libpldmresponder/file_io_type_pcie.cpp

```diff
@@ -0,0 +1,148 @@
+#include "file_io_type_pcie.hpp"
+
+#include "libpldm/base.h"
+#include "oem/ibm/libpldm/file_io.h"
+
+#include "common/utils.hpp"
+#include "host-bmc/dbus/custom_dbus.hpp"
+
+#include <stdint.h>
+
+#include <iostream>
+
+namespace pldm
+{
+namespace responder
+{
+
+static constexpr auto pciePath = "/var/lib/pldm/pcie-topology/";
+constexpr auto topologyFile = "topology";
+constexpr auto cableInfoFile = "cableinfo";
+
+namespace fs = std::filesystem;
+std::map<uint16_t, bool> PCIeInfoHandler::receivedFiles;
+
+int PCIeInfoHandler::writeFromMemory(
+    uint32_t offset, uint32_t length, uint64_t address,
+    oem_platform::Handler* /*oemPlatformHandler*/)
+{
+    if (!fs::exists(pciePath))
+    {
+        fs::create_directories(pciePath);
+        fs::permissions(pciePath,
+                        fs::perms::others_read | fs::perms::owner_write);
+    }
+
+    fs::path infoFile(fs::path(pciePath) / topologyFile);
+    if (infoType == PLDM_FILE_TYPE_CABLE_INFO)
+    {
+        infoFile = (fs::path(pciePath) / cableInfoFile);
+    }
+
+    std::ofstream pcieData(infoFile, std::ios::out | std::ios::binary);
+    if (!pcieData)
+    {
+        std::cerr << "PCIe Info file creation error " << std::endl;
+        return -1;
```

Comment:
we dont have -1 as a return code in pldm , it can only take one of these values.

	PLDM_SUCCESS = 0x00,
	PLDM_ERROR = 0x01,
	PLDM_ERROR_INVALID_DATA = 0x02,
	PLDM_ERROR_INVALID_LENGTH = 0x03,
	PLDM_ERROR_NOT_READY = 0x04,
	PLDM_ERROR_UNSUPPORTED_PLDM_CMD = 0x05,
	PLDM_ERROR_INVALID_PLDM_TYPE = 0x20,
	PLDM_INVALID_TRANSFER_OPERATION_FLAG = 0x21

Reply:
Done